mcp/mastra/verticals/web-research/tools.ts

```diff
@@ -1,138 +1,27 @@
-      max_results: inputData.maxResults,
-      include_domains: [],
-      exclude_domains: [],
-    };
-
-    const response = await fetch(url, {
-      method: 'POST',
-      headers: {
-        'Content-Type': 'application/json',
-      },
-      body: JSON.stringify(requestBody),
-    });
-
-    if (!response.ok) {
-      throw new Error(`Failed to search web: ${response.statusText}`);
-    }
-
-    const data = await response.json();
-
-    return {
-      answer: data.answer,
-      results: data.results || [],
-      images: data.images || [],
-    };
-  },
-});
-
-// Tool to get HTML content from URLs using Tavily extract API
-export const getHtmlContent = createTool({
-  id: 'getHtmlContent',
-  description: 'Get the HTML contents of a particular web page URL.',
-  inputSchema: z.object({
-    url: z.string().describe('The URL to fetch HTML content from'),
-  }),
-  outputSchema: z.object({
-    url: z.string(),
-    rawContent: z.string().describe('The extracted text content from the HTML page'),
-  }),
-  execute: async (inputData) => {
-    const apiKey = getApiKey();
-    const url = 'https://api.tavily.com/extract';
-
-    const requestBody = {
-      api_key: apiKey,
-      urls: [inputData.url],
-    };
-
-    const response = await fetch(url, {
-      method: 'POST',
-      headers: {
-        'Content-Type': 'application/json',
-      },
-      body: JSON.stringify(requestBody),
-    });
-
-    if (!response.ok) {
-      throw new Error(`Failed to extract HTML content: ${response.statusText}`);
-    }
-
-    const data = await response.json();
-
-    // Tavily extract returns an object with results keyed by URL
-    const results = data.results || [];
-    const result = results[0] || { url: inputData.url, raw_content: '' };
-
-    return {
-      url: result.url || inputData.url,
-      rawContent: result.raw_content || '',
-    };
-  },
-});
-
-// Export all tools together for convenience
-export const webResearchTools = {
-  searchWeb,
-  getHtmlContent,
-};
+/**
+ * Web Research Tools
+ *
+ * NOTE: Tavily-based tools have been PERMANENTLY DISABLED in favor of Google Search Grounding.
+ * Google Search Grounding is integrated directly into the agent's model (gemini-2.0-flash-exp)
+ * and provides real-time web search with automatic source citations without needing separate tools.
+ *
+ * The web research agent now uses Google Search grounding through providerOptions:
+ * {
+ *   google: {
+ *     groundingConfig: {
+ *       googleSearchRetrieval: {}
+ *     }
+ *   }
+ * }
+ *
+ * This provides:
+ * - Real-time web search integrated into model responses
+ * - Automatic source citations and grounding metadata
+ * - No separate API calls or tools needed
+ * - Better integration between search and response generation
+ *
+ * The Tavily API key is no longer required for this agent.
+ */
+
+// No tools are exported - the agent uses Google Search grounding instead
+export const webResearchTools = {};
```

Comment:
You are forgetting to add the needed grounding tool: google.tools.googleSearch({})